client/src/partials/budget/new/new_budget.js

```diff
@@ -0,0 +1,368 @@
+        b.freeze = false;
+        b.editing = false;
+        total += Number(b.budget);
+      });
+      session.totalBudget = total;
+
+      if (session.found) {
+        recompute();
+      }
+      else {
+        session.no_data_msg = $translate.instant('BUDGET.EDIT.DATA_NOT_FOUND')
+          .replace('(acct)', session.account.account_txt)
+          .replace('(fyname)', session.fiscal_year.fiscal_year_txt);  
+      }
+    }
+
+    // Initialize editing the selected account
+    function submitAccount(newAccount) {
+      if (newAccount) { 
+        session.account = newAccount;
+        dependencies.account.query.where = ['account.id=' + newAccount.id];
+        dependencies.budgets.query.where = ['period.fiscal_year_id=' + session.fiscal_year.id, 'AND',
+              'budget.account_id=' + session.account.id, 'AND',
+              'period.fiscal_year_id=' + session.fiscal_year.id, 'AND',
+              'period.period_number<>0'];
+        // NOTE: Restricting the periods to the selected fiscal year
+        //       automatically limits the budget items to ones for this
+        //       enterprise since the specific FY is tied to a particular
+        //       enterprise.
+        dependencies.periods.query.where = ['period.fiscal_year_id=' + session.fiscal_year.id, 'AND',
+        'period.period_number<>0'];
+        validate.refresh(dependencies, ['account', 'budgets', 'periods'])
+        .then(startup);
+        session.mode = 'edit';
+      }
+    }
+
+    function accountWhere() {
+      // only unlocked income/expense accounts
+      // Allow accounts of class 1 and 2 used in budget
+      return [['account.account_type_id IN (1,4)','OR',['account.classe IN (1,2,5)','AND','account.is_used_budget=1']], 'AND', 'account.locked=0' ];
+    }
+
+    function restartSearch() {
+      session.mode = 'configure';
+      session.option = 'manual';
+      session.autoAdjust = false;
+    }
+
+    // BUDGET: 
+    //   `id` int not null auto_increment,
+    //   `account_id` int unsigned not null default '0',
+    //   `period_id` mediumint unsigned not null,
+    //   `budget` decimal(10,4) unsigned,
+
+    function createBudget() {
+      var newBudgets = [];
+      $scope.periods.data.forEach(function (per, index) {
+      newBudgets.push({'account_id' : session.account.id,
+       'period_id' : per.id,
+       'budget' : 0.0});
+      });
+
+      connect.post('budget', newBudgets, ['id'])
+      .then(function () {
+        messenger.success($translate.instant('BUDGET.EDIT.CREATE_OK'));
+        submitAccount(session.account);
+      })
+      .catch(function (err) {
+      messenger.danger($translate.instant('BUDGET.EDIT.CREATE_FAIL'));
+      console.log(err);
+      });
+    }
+
+    function updateBudget() {
+      // Save the budget data for all the periods
+      var dbPromises = [];
+      $scope.budgets.data.forEach(function (bud) {
+      dbPromises.push( 
+      connect.put('budget', 
```

Comment:
Here, you are generating a loop of HTTP requests.  This is exactly the problems we reported on #624, #679 and others.

A better model is to create a server-side route that receives one HTTP request and send all the data at once to that endpoint.  It can then validate the data and insert it into the database.